File: quickactionview/src/main/java/com/commit451/quickactionview/ConfigHelper.java
package com.commit451.quickactionview;

import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.Typeface;
import android.graphics.drawable.Drawable;
import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;

/**
 * Determines which config to get values from, based on the state of each config.
 */
class ConfigHelper {
    private Action.Config mActionConfig;
    private QuickActionView.Config mQuickActionViewConfig;

    public ConfigHelper(@NonNull Action.Config actionConfig, @NonNull QuickActionView.Config quickActionViewConfig) {
        mActionConfig = actionConfig;
        mQuickActionViewConfig = quickActionViewConfig;
    }

    public Drawable getTextBackgroundDrawable(Context context) {
        if (mActionConfig != null && mActionConfig.getTextBackgroundDrawable(context) != null) {
            return mActionConfig.getTextBackgroundDrawable(context);
        }
        return mQuickActionViewConfig.getTextBackgroundDrawable(context);
    }


    @ColorInt
    public int getTextColor() {
        if (mActionConfig != null && mActionConfig.getTextColor() != 0) {
            return mActionConfig.getTextColor();
        }
        return mQuickActionViewConfig.getTextColor();
    }


    public ColorStateList getBackgroundColorStateList() {
        if (mActionConfig != null && mActionConfig.getBackgroundColorStateList() != null) {
            return mActionConfig.getBackgroundColorStateList();
        }
        return mQuickActionViewConfig.getBackgroundColorStateList();
    }

    public Typeface getTypeface() {
        return mQuickActionViewConfig.getTypeface();
    }

    public int getTextSize() {
        return mQuickActionViewConfig.getTextSize();
    }

    public int getTextPaddingTop() {
        return mQuickActionViewConfig.getTextPaddingTop();
    }

    public int getTextPaddingBottom() {
        return mQuickActionViewConfig.getTextPaddingBottom();
    }

    public int getTextPaddingLeft() {
        return mQuickActionViewConfig.getTextPaddingLeft();
    }

    public int getTextPaddingRight() {
        return mQuickActionViewConfig.getTextPaddingRight();
    }

}

File: quickactionview/src/main/java/com/commit451/quickactionview/ConfigHelper.kt
package com.commit451.quickactionview

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.Typeface
import android.graphics.drawable.Drawable
import androidx.annotation.ColorInt

/**
 * Determines which config to get values from, based on the state of each config.
 */
class ConfigHelper(
        private val actionConfig: Action.Config?,
        private val quickActionViewConfig: QuickActionView.Config
) {

    val textColor: Int
        @ColorInt
        get() = if (actionConfig != null && actionConfig.textColor != 0) {
            actionConfig.textColor
        } else quickActionViewConfig.textColor


    val backgroundColorStateList: ColorStateList
        get() = if (actionConfig?.backgroundColorStateList != null) {
            actionConfig.backgroundColorStateList
        } else quickActionViewConfig.backgroundColorStateList

    val typeface: Typeface?
        get() = quickActionViewConfig.typeface

    val textSize: Int
        get() = quickActionViewConfig.textSize

    val textPaddingTop: Int
        get() = quickActionViewConfig.textPaddingTop

    val textPaddingBottom: Int
        get() = quickActionViewConfig.textPaddingBottom

    val textPaddingLeft: Int
        get() = quickActionViewConfig.textPaddingLeft

    val textPaddingRight: Int
        get() = quickActionViewConfig.textPaddingRight

    fun getTextBackgroundDrawable(context: Context): Drawable? {
        return if (actionConfig?.getTextBackgroundDrawable(context) != null) {
            actionConfig.getTextBackgroundDrawable(context)
        } else quickActionViewConfig.getTextBackgroundDrawable(context)
    }

}
